lib/pages/search/server_search_controller.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/pages/search/search_debouncer_mixin.dart';
+import 'package:fluffychat/domain/usecase/search/server_search_interactor.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+import 'package:fluffychat/domain/app_state/search/server_search_state.dart';
+
+class ServerSearchController with SearchDebouncerMixin {
+  final _serverSearchInteractor = getIt.get<ServerSearchInteractor>();
+
+  final serverSearchResultNotifier = ValueNotifier<List<MatrixEvent?>?>([]);
```

Comment:
```suggestion
  final serverSearchStateNotifier = ValueNotifier<Either<Failure, Success>>(const Right(ServerSearchInitial()));
```

Reply:
i prefer `List<MatrixEvent?>`, so that i don't have to move the logic of `Either<Failure, Success>>` to the UI widget